entity-service/src/main/kotlin/com/egm/stellio/entity/authorization/Neo4jAuthorizationRepository.kt

```diff
@@ -227,4 +231,50 @@ class Neo4jAuthorizationRepository(
 
         return neo4jClient.query(matchQuery).bindAll(parameters).run().counters().nodesDeleted()
     }
+
+    fun getGroupEntity(groupId: URI): Group {
+        val query =
+            """
+            MATCH (group:Entity { id: ${'$'}groupId })-[:HAS_VALUE]->(p:Property { name:"$NGSILD_NAME_PROPERTY" })
+            RETURN p.value as groupName
+            """.trimIndent()
+
+        val parameters = mapOf(
+            "groupId" to groupId.toString()
+        )
+
+        val result = neo4jClient.query(query).bindAll(parameters).fetch().one()
+
+        return result.map {
+            Group(
+                id = groupId,
+                type = listOf(GROUP_TYPE),
+                name = it["groupName"] as String
+            )
+        }.get()
+    }
+
+    fun getGroupsAdmin(groupMembership: Set<URI>): List<Group> {
+        val query =
+            """
+            MATCH (group:Entity)-[:HAS_VALUE]->(p:Property { name:"$NGSILD_NAME_PROPERTY" })
```

Comment:
you should specify the label of the group ... here you will get all entities that have a ``name`` attribute, that makes a lot!